pilot/cmd/pilot-agent/main.go

```diff
@@ -516,6 +546,16 @@ var (
 	}
 )
 
+func getGcpProjectNumber() string {
+	if platform.IsGCP() {
+		md := platform.NewGCP().Metadata()
+		if projNum, found := md[platform.GCPProjectNumber]; found {
```

Comment:
this isn't a free request, why do we have to call metadata server again instead of reusing?

Reply:
The existing flow of using the metadata is like this: in main.go of Pilot agent, 
```watcher := envoy.NewWatcher(tlsCertsToWatch, agent.Restart)``` ->
```agent.Restart``` creates a goroutine by -> ```a.runWait(config, epoch, abortCh) pkg/envoy/agent.go``` ->
```runWait()``` calls ```a.proxy.Run() pkg/envoy/proxy.go```-> 
```a.proxy.Run() ``` calls ```bootstrap.New(bootstrap.Config{}).CreateFileForEpoch(epoch)``` ->
```CreateFileForEpoch()``` calls ```WriteTo() pkg/bootstrap/instance.go```->
 ```WriteTo()``` calls ```i.toTemplateParams() pkg/bootstrap/config.go```. 

toTemplateParams() returns a map of metadata, and CreateFileForEpoch() converts metadata to a file content and write into proxy bootstrap config file. 
It seems that Pilot agent cannot access the map and get project number directly. Are there other use cases of metadata server in Pilot agent that I missed?